query-engine/connectors/mongodb-query-connector/src/filter.rs

```diff
@@ -36,872 +37,1070 @@ pub(crate) struct MongoRelationFilter {
+                    _ => {
+                        let matches = vals
+                            .into_iter()
+                            .map(|val| self.regex_match(&field_name, field, "^", val, "$", true))
+                            .collect::<crate::Result<Vec<_>>>()?;
 
-                Ok(doc! { "$or": matches })
-            }
-        },
-        ScalarCondition::NotIn(vals) => {
-            let matches = vals
-                .into_iter()
-                .map(|val| regex_match(&field_name, field, "^", val, "$", true).map(|doc| doc! { "$not": doc }))
-                .collect::<crate::Result<Vec<_>>>()?;
+                        Ok(doc! { "$or": matches })
+                    }
+                },
+                ConditionListValue::FieldRef(field_ref) => Ok(render_some(
+                    (self.prefix(), &field_ref).into_bson()?,
+                    "elem",
+                    self.regex_match(&Bson::from("$$elem"), field, "^", field, "$", true)?,
+                    true,
+                )),
+            },
+            ScalarCondition::NotIn(vals) => match vals {
+                ConditionListValue::List(vals) => {
+                    let matches = vals
+                        .into_iter()
+                        .map(|val| {
+                            self.regex_match(&field_name, field, "^", val, "$", true)
+                                .map(|rgx_doc| doc! { "$not": rgx_doc })
+                        })
+                        .collect::<crate::Result<Vec<_>>>()?;
+
+                    Ok(doc! { "$and": matches })
+                }
+                ConditionListValue::FieldRef(field_ref) => Ok(render_every(
+                    (self.prefix(), &field_ref).into_bson()?,
+                    "elem",
+                    self.regex_match(&Bson::from("$$elem"), field, "^", field, "$", true)
+                        .map(|rgx_doc| doc! { "$not": rgx_doc })?,
+                    true,
+                )),
+            },
+            ScalarCondition::IsSet(is_set) => Ok(render_is_set(&field_name, is_set)),
+            ScalarCondition::JsonCompare(_) => Err(MongoError::Unsupported(
+                "JSON filtering is not yet supported on MongoDB".to_string(),
+            )),
+            ScalarCondition::Search(_, _) | ScalarCondition::NotSearch(_, _) => Err(MongoError::Unsupported(
+                "Full-text search is not supported yet on MongoDB".to_string(),
+            )),
+        }?;
+
+        let filter_doc = if !is_set_cond {
+            exclude_undefineds(&field_name, self.invert_undefined_exclusion(), filter_doc)
+        } else {
+            filter_doc
+        };
+
+        let filter_doc = if let Some(field_ref) = &field_ref {
+            exclude_undefineds(
+                self.prefixed_field_ref(field_ref)?,
+                self.invert_undefined_exclusion(),
+                filter_doc,
+            )
+        } else {
+            filter_doc
+        };
```

Comment:
Fix for https://github.com/prisma/prisma/issues/14664 ; We do the exact same thing for other types of filters, this was an oversight on the original implementation.